Bottom: 299a341225444040a934ba533c4ede1458ed8427
Top:    8c2a5deadfb074d278870f46e25a0e3ac515d8d0
Author: Robert Collins <robertc@robertcollins.net>
Date:   2013-02-23 22:48:44 +1300

Refresh of copy-stream

---

diff --git a/testtools/testresult/real.py b/testtools/testresult/real.py
index 72aad0f..7d264d8 100644
--- a/testtools/testresult/real.py
+++ b/testtools/testresult/real.py
@@ -385,10 +385,6 @@ class CopyStreamResult(StreamResult):
         super(CopyStreamResult, self).stopTestRun()
         domap(methodcaller('stopTestRun'), self.targets)
 
-    def file(self, *args, **kwargs):
-        super(CopyStreamResult, self).file(*args, **kwargs)
-        domap(methodcaller('file', *args, **kwargs), self.targets)
-
     def status(self, *args, **kwargs):
         super(CopyStreamResult, self).status(*args, **kwargs)
         domap(methodcaller('status', *args, **kwargs), self.targets)
diff --git a/testtools/tests/test_testresult.py b/testtools/tests/test_testresult.py
index 3f5a247..8d19ccb 100644
--- a/testtools/tests/test_testresult.py
+++ b/testtools/tests/test_testresult.py
@@ -586,23 +586,16 @@ class TestCopyStreamResultCopies(TestCase):
         self.assertThat(self.targets,
             AllMatch(Equals([('startTestRun',), ('stopTestRun',)])))
 
-    def test_file(self):
-        self.result.startTestRun()
-        now = datetime.datetime.now(utc)
-        self.result.file("foo", "bar", eof=True, mime_type="text/json",
-            test_id="id", route_code='abc', timestamp=now)
-        self.assertThat(self.targets,
-            AllMatch(Equals([('startTestRun',),
-                ('file', 'foo', 'bar', True, 'text/json', 'id', 'abc', now)])))
-
     def test_status(self):
         self.result.startTestRun()
         now = datetime.datetime.now(utc)
         self.result.status("foo", "success", test_tags=set(['tag']),
-            runnable=False, route_code='abc', timestamp=now)
+            runnable=False, file_name="foo", file_bytes=b'bar', eof=True,
+            mime_type="text/json", route_code='abc', timestamp=now)
         self.assertThat(self.targets,
             AllMatch(Equals([('startTestRun',),
-                ('status', 'foo', 'success', set(['tag']), False, 'abc', now)
+                ('status', 'foo', 'success', set(['tag']), False, "foo",
+                 b'bar', True, "text/json", 'abc', now)
                 ])))
